narwhals/functions.py

```diff
@@ -29,6 +31,9 @@
     from narwhals.series import Series
     from narwhals.typing import DTypes
 
+    class SupportsPyCapsule(Protocol):
+        def __arrow_c_stream__(self) -> Any: ...
```

Comment:
This isn't the correct type hint: https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html#protocol-typehints